common/hyperscaler/rules/rules_service.go

```diff
@@ -108,6 +108,20 @@ func (rs *RulesService) parse(rulesConfig *RulesConfig) *ParsingResults {
 	return results
 }
 
+// MatchProvisioningAttributes finds the matching rule for the given provisioning attributes and provide values needed to create labels, which must be used to find proper secret binding.
+func (rs *RulesService) MatchProvisioningAttributes(provisioningAttributes *ProvisioningAttributes) (Result, bool) {
+	var result Result
+	found := false
+	for _, parsingResult := range rs.Parsed.Results {
+		if parsingResult.Rule.Matched(provisioningAttributes) {
```

Comment:
Are we sure there is only one rule matching, or the first we find is the proper one?

Reply:
the original Match method works the same